### ISSUE TITLE
nz-avatar scale is incorredct

### ISSUE THREAD

**Issue by HaidarVladyslav** [NG-ZORRO/ng-zorro-antd#8363]
### Reproduction link
[https://stackblitz.com/edit/stackblitz-starters-t4ldds?file=sample-application%2Fsrc%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/stackblitz-starters-t4ldds?file=sample-application%2Fsrc%2Fapp%2Fapp.component.ts)

### Steps to reproduce
SORRY, I DONT HAVE REPRO PUBLICLY, Link to stakblitz makes no sense
https://postimg.cc/HV3qc4Hq
I have nz-avatar and its size is incorrect. Scale is zero. Actually I don't know why. I migrated v14 to v16 and worked directly with components in storybook and didn't see any issues. But once I just started to work with previously existing app, I can see this issue. It also happens if I have sidemenu that is closed by default (that's regular Angular component that uses nz-avatar). And interesting thing that if I have sidebar as default nz-sider, its size is correct, but once I reduce screensize, I operate with closed/opened states and can see this issue. But, if I firstly was on large screen, then reduced it to tablet, so the size of Text inside nz-avatar is correct, but if I reload page as tablet view, can see this issue...

### What is expected?
Expected - size is set correctly

### What is actually happening?
https://postimg.cc/HV3qc4Hq

| Environment | Info |
|---|---|
| ng-zorro-antd | 16.0.0 |
| Browser | Chrome |

**Comment by HaidarVladyslav** [comment]
Actually I just reproduced this on Stakblitz with v17. I think it can be common issue

**Comment by ParsaArvanehPA** [comment]
Hi @HaidarVladyslav,
This is a possible fix for your issue #8365.

**Comment by ParsaArvanehPA** [comment]
In the meantime, you can use this hack to get your problem solved:

component.html:
``` ts
<nz-modal
  [(nzVisible)]="isVisible"
  nzTitle="The first Modal"
  (nzOnCancel)="handleCancel()"
  (nzOnOk)="handleOk()"
  (nzAfterOpen)="reScaleAvatar()"
>
  <ng-container *nzModalContent>
    <nz-avatar nzText="U S" #avatarComponent></nz-avatar>
  </ng-container>
</nz-modal>
```


component.ts:
```ts
@ViewChild('avatarComponent')
public avatarComponent!: NzAvatarComponent;

public reScaleAvatar(): void {
  this.avatarComponent['notifyCalc']();
}
```

Not that I am bracket syntax to call `notifyCalc` because it is a private property inside avatar component and cannot be accessed from outside.

**Comment by HaidarVladyslav** [comment]
> Hi @HaidarVladyslav, This is a possible fix for your issue #8365.

I also tried that. But then other examples are broken. You can check existing 'Change Text' example, its position is incorrect.
I also tried to make 'setTimeout' with bigger value (50ms), but it works for my case. But in general we don't have to do like this.

Calling private method to rerender avatar-component also looks for me non-proper way

But thanks for your quick response!

**Comment by HaidarVladyslav** [comment]
@ParsaArvanehPA and about your temporary solution. Let's imagine I have avatar-component inside my custom component. My custom component is inside another component. And only then I have modal. This nesting to call 'nzAfterOpen' is strange. And if I have different usages of avatar-component across the app or across multiple apps, I need to call this method everywhere..

**Comment by ParsaArvanehPA** [comment]
> > Hi @HaidarVladyslav, This is a possible fix for your issue #8365.
> 
> I also tried that. But then other examples are broken. You can check existing 'Change Text' example, its position is incorrect. I also tried to make 'setTimeout' with bigger value (50ms), but it works for my case. But in general we don't have to do like this.
> 
> Calling private method to rerender avatar-component also looks for me non-proper way
> 
> But thanks for your quick response!

So you are saying you checked out my branch and there were some parts not working properly? because that change text part seems to be working fine in #8365

![image](https://github.com/NG-ZORRO/ng-zorro-antd/assets/62149413/55c4d0d8-474c-41d3-90ed-8d902c89ea08)

**Comment by ParsaArvanehPA** [comment]
> @ParsaArvanehPA and about your temporary solution. Let's imagine I have avatar-component inside my custom component. My custom component is inside another component. And only then I have modal. This nesting to call 'nzAfterOpen' is strange. And if I have different usages of avatar-component across the app or across multiple apps, I need to call this method everywhere..

Yeah it is just a hack, best to wait for the solution to be merged.

**Comment by HaidarVladyslav** [comment]
> > > Hi @HaidarVladyslav, This is a possible fix for your issue #8365.
> > 
> > 
> > I also tried that. But then other examples are broken. You can check existing 'Change Text' example, its position is incorrect. I also tried to make 'setTimeout' with bigger value (50ms), but it works for my case. But in general we don't have to do like this.
> > Calling private method to rerender avatar-component also looks for me non-proper way
> > But thanks for your quick response!
> 
> So you are saying you checked out my branch and there were some parts not working properly? because that change text part seems to be working fine in #8365
> 
> ![image](https://private-user-images.githubusercontent.com/62149413/298117580-55c4d0d8-474c-41d3-90ed-8d902c89ea08.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2Nzk1NzUsIm5iZiI6MTcwNTY3OTI3NSwicGF0aCI6Ii82MjE0OTQxMy8yOTgxMTc1ODAtNTVjNGQwZDgtNDc0Yy00MWQzLTkwZWQtOGQ5MDJjODllYTA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE5VDE1NDc1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1MmIxMTg3NzZlNDQxMGE2MjdlZWFhMTY2ZGY5ODM0ZWQyNDY2YmEyZDYyYzA3MjMzN2FkNzc0YjZhNDgwNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IYp0isxDqqx4PAgjN65q3p8GHyOcs82JvHHg-h-F_5s)

yes, please, look here - I just was spamming 'Change Text', and it's out of container
![image](https://github.com/NG-ZORRO/ng-zorro-antd/assets/93485656/fbb61791-6788-463d-9ad7-ee3bd4b0a4cb)
